XenAdmin/Wizards/PatchingWizard/PatchingWizard_ModePage.cs

```diff
@@ -165,6 +185,29 @@ private void ManualRadioButton_CheckedChanged(object sender, EventArgs e)
             UpdateEnablement();
         }
 
+        private bool AnyPoolForbidsAutoRestart()
+        {
+            var poolForbidRestartKey = "XenCenter.CustomFields.hci-forbid-update-auto-restart";
+            foreach (var server in SelectedServers)
+            {
+                var pool = Helpers.GetPoolOfOne(server.Connection);
+                var poolOtherConfig = Helpers.GetOtherConfig(pool);
+                if (poolOtherConfig.ContainsKey(poolForbidRestartKey) &&
+                    poolOtherConfig[poolForbidRestartKey].ToLowerInvariant().Equals(bool.TrueString.ToLowerInvariant()))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private bool PatchRequiresReboot()
+        {
+            return !textBoxLog.Text.Equals(Messages.PATCHINGWIZARD_MODEPAGE_NOACTION);

```

Comment:
If you are already making changes here, I'd suggest this should be a getter property. For the string, you could invent another new property to avoid having an out parameter.

Also, it's a good practice to name such bool properties to start with "Is". "IsRebootRequired" or similar.